src/Scope-RedPitaya_STEMlab/main.py

```
@@ -65,226 +66,483 @@ def __init__(self):
+            self.port.write(
+                "ACQ:SOUR{0}:DATA:OLD:N? {1}".format(
+                    self.channels[i], self.read_samples
+                )
+            )
             self.buffer = self.port.read()
             try:
-                self.data = list(map(float, self.buffer.strip('{}\n\r').replace("  ", "").split(',')))
+                self.data = list(
+                    map(float, self.buffer.strip("{}\n\r").replace("  ", "").split(","))
+                )
             except:
```

Ruff (0.16.1)

[error] Do not use bare `except`

(E722)